### PR TITLE
Do not set properties if the value is equal to what is already set

### PR DESCRIPTION
**Pull request by etbyrd**
Fixes: https://github.com/dotnet/project-system/issues/4519

There are a couple of scenarios where this is relevant:

1) As in https://github.com/dotnet/project-system/issues/4519, values are set in response to other values being set - even if they are empty.
2) If a property is edited, it will be marked as dirty. If it is edited again before a save back to its current value and then saved, it will still be marked as dirty and the value will be set. If this property was empty, it will create a new empty property in the .*proj file. If the property was being pulled from some default value (like PackageId), then a property will be added with the default value. 

Both of these will clutter up the .*proj file.

Ideally, I would like to never have empty properties - but there might be some legitimate uses for having them so it is out of the scope for this PR.

### CODE REVIEW

**Comment by jmarolf** [comment]
close/reopen to get tests to run

File: src/Microsoft.VisualStudio.AppDesigner/PropPages/PropertyControlData.vb [hunk 1]
```diff
@@ -1714,13 +1714,20 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
 
                 _valueChangedWasFired = False
                 Try
-                    'Go ahead and do the SetValue.  It will throw if there's an exception
-                    '  (other than cancel/checkout exceptions).
-                    Descriptor.SetValue(Component, Value)
+                    Dim _didNotSetEqualValue = False
```
[the remark this answers is not in the document]
Comment (etbyrd) [reply src/Microsoft.VisualStudio.AppDesigner/PropPages/PropertyControlData.vb]: Good point - not sure what I was thinking.

**Comment by davkean** [comment]
This a project system concern, not a UI concern. We have APIs that will repro this exact problem that don't need to go through the AppDesigner, we also situations where this occurs in property browser (clear Custom Tool property on an item and you will see the same thing). This is a CPS behavior and we should change it at the root, their behavior is based around C++ not our (keep a clean project file) managed scenarios.

**Comment by etbyrd** [comment]
@Pilchie This shouldn't change the behavior on when it throws/doesn't throw in the else clause. I am not 100% why they had that behavior, but this should keep it intact.

**Comment by Pilchie** [comment]
Agree with @davkean here.  This doesn't seem like the right place to be addressing this.

**Comment by etbyrd** [comment]
Closing - we will need to change this in CPS

**Pull request closed without merging**